### ISSUE TITLE
meta.json is not removed when the container process exits.

### ISSUE THREAD

**Issue by fahedouch** [containerd/nerdctl#1829]
### Description

container `Poststop` hook is not working as expected: `meta.json` is not removed when the container process exits.

### Steps to reproduce the issue

1. nerdctl run alpine
2. `/var/lib/nerdctl/<ADDRHASH>/etchosts/<NS>/<ID>/meta.json` is still present

### Describe the results you received and expected

`/var/lib/nerdctl/<ADDRHASH>/etchosts/<NS>/<ID>/meta.json` should be removed when container with id  `ID` exists

### What version of nerdctl are you using?

latest

### Are you using a variant of nerdctl? (e.g., Rancher Desktop)

None

**Comment by fahedouch** [comment]
container `Poststop` is executed in the Runtime Namespace and not the container ns

**Comment by apostasie** [comment]
Hey folks.

Wondering if we do really want this.
`meta.json` will get deleted when the container gets _removed_ (postStop).

Is removing at `stop` really bring anything valuable to the table?

**Comment by apostasie** [comment]
@fahedouch ^

**Comment by apostasie** [comment]
@AkihiroSuda @fahedouch shall we close this?
I do not think we should remove it when the container stops. It is removed when the container is removed though.

**Comment by AkihiroSuda** [comment]
We may remove it when the container exits, but it’s not a critical requirement

**Comment by apostasie** [comment]
> We may remove it when the container exits, but it’s not a critical requirement

I do not think it is feasible, as we do not get ocihooks events for container stop/start.
Generally speaking, we cannot ensure we do things when containers stop outside of us explicitly calling `stop`.
